include/seqan3/alignment/pairwise/align_pairwise.hpp

```diff
@@ -38,10 +38,92 @@
 namespace seqan3
 {
 
+/*!\brief Computes a pairwise alignment over a sequence pair or a range of sequence pairs.
+ * \ingroup pairwise_alignment
+ *
+ * \tparam sequence_t         The type of sequence pairs. Either a type modelling seqan3::tuple_like_concept
+ *                            with a tuple size of exactly two or std::ranges::InputRange whose value type models the
+ *                            before mentioned tuple concept.
+ * \tparam alignment_config_t The type of the alignment configuration; must be a seqan3::configuration.
+ * \param[in] seq             A tuple with two sequences or an input range over such tuples.
+ * \param[in] config          A configuration object containing \ref alignment_configuration
+ *                            "alignment configurations".
+ * \return Depending on the configuration, the return type of align_pairwise is either `void` if the alignment was
+ * configured to call a delegate on each hit or a seqan3::alignment_range over the alignment results.
+ *
+ * \details
+ *
+ * This function computes the pairwise alignment of a sequence pair or a range of sequence pairs.
+ * The type of the elements stored in the tuples must model std::ranges::ForwardRange.
+ * The value category of these elements is either an lvalue or if the passed tuple or range is an lvalue, xvalues are
+ * also allowed as tuple elements.
```

Comment:
What is the intention of this sentence? Which input are not allowed, can you give me an example?
And does the user understand this?

Reply:
Something like std::tie(rng1, rng2) gives you a prvalue with lvalue references wrapped inside. 
Something like std::tuple{"ACGT", "ACGT"} is only allowed, if the tuple itself is not a xvalue. That is because accessing the values of the tuple via get inside the alignment will return lvalues but they become dangling references after the invocation of align_pairwise. But I agree that the sentence might be confusing.